warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1286,7 +1285,6 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
                 // if we now have an unexecutable tree because of delayed
                 // predicates, then remove delayed predicates as needed and
                 // reexpand
-                List<String> debugOutput = null;
                 if (log.isDebugEnabled()) {
                     debugOutput = new ArrayList<>(32);
                 }
```

Comment:
Is it intended to re-create the debugOutput array list here, now that it's created above?

Reply:
Yeah, the Executability determination visitor needs the debug output ( if enabled )...so I've moved the creation above as I use it to verify executability. It then gets re-created for the next round.